### ISSUE TITLE
V. 2.3.0 Installer fails (likely as MSI install path has no value)

### ISSUE THREAD

**Issue by OttoBoll** [Microsoft/IIS.Administration#291]
Trying to install IISAdministrationSetup.exe (2.3.0) throws a "Fatal Error", error log passes pre-req & planning checks with 0x0.
Here's the following lines:

[1310:0624][2019-11-26T13:28:45]i300: Apply begin
[1310:0624][2019-11-26T13:28:45]i010: Launching elevated engine process.
[1310:0624][2019-11-26T13:28:47]i011: Launched elevated engine process.
[1310:0624][2019-11-26T13:28:47]i012: Connected to elevated engine.
[1074:07F8][2019-11-26T13:28:47]i358: Pausing automatic updates.
[1074:07F8][2019-11-26T13:28:47]i359: Paused automatic updates.
[1074:07F8][2019-11-26T13:28:47]i360: Creating a system restore point.
[1074:07F8][2019-11-26T13:28:47]i362: System restore disabled, system restore point not created.
[1074:07F8][2019-11-26T13:28:47]i370: Session begin, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{5911cf72-9c50-4662-a712-28a4c2f8b599}, options: 0x7, disable resume: No
[1074:07F8][2019-11-26T13:28:47]i000: Caching bundle from: 'C:\Users\[redacted]\AppData\Local\Temp\2\{336E5AD6-7975-433F-82B0-15DF03F95FB7}\.be\IISAdministrationSetup.exe' to: 'C:\ProgramData\Package Cache\{5911cf72-9c50-4662-a712-28a4c2f8b599}\IISAdministrationSetup.exe'
[1074:07F8][2019-11-26T13:28:47]i320: Registering bundle dependency provider: {5911cf72-9c50-4662-a712-28a4c2f8b599}, version: 2.3.0.0
[1074:07F8][2019-11-26T13:28:47]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{5911cf72-9c50-4662-a712-28a4c2f8b599}, resume: Active, restart initiated: No, disable resume: No
[1074:100C][2019-11-26T13:28:47]i304: Verified existing payload: iis_administration_msi at path: C:\ProgramData\Package Cache\{51B98E36-BBAC-4549-8460-DBEFFE4FDD06}v2.3.0\packages\iis_administration.msi.
[1074:07F8][2019-11-26T13:28:47]i323: Registering package dependency provider: {51B98E36-BBAC-4549-8460-DBEFFE4FDD06}, version: 2.3.0, package: iis_administration_msi
[1074:07F8][2019-11-26T13:28:47]i301: Applying execute package: iis_administration_msi, action: Install, path: C:\ProgramData\Package Cache\{51B98E36-BBAC-4549-8460-DBEFFE4FDD06}v2.3.0\packages\iis_administration.msi, arguments: ' ARPSYSTEMCOMPONENT="1" MSIFASTINSTALL="7" INSTALLFOLDER="" VSEXTUI="1" DEFAULT_CORS="true"'
[1074:07F8][2019-11-26T13:28:54]e000: Error 0x80070643: Failed to install MSI package.
[1074:07F8][2019-11-26T13:28:54]e000: Error 0x80070643: Failed to execute MSI package.
[1310:0624][2019-11-26T13:28:54]e000: Error 0x80070643: Failed to configure per-machine MSI package.
[1310:0624][2019-11-26T13:28:54]i319: Applied execute package: iis_administration_msi, result: 0x80070643, restart: None
[1310:0624][2019-11-26T13:28:54]e000: Error 0x80070643: Failed to execute MSI package.
[1074:07F8][2019-11-26T13:28:54]i372: Session end, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{5911cf72-9c50-4662-a712-28a4c2f8b599}, resume: None, restart: None, disable resume: No
[1074:07F8][2019-11-26T13:28:54]i330: Removed bundle dependency provider: {5911cf72-9c50-4662-a712-28a4c2f8b599}
[1074:07F8][2019-11-26T13:28:54]i352: Removing cached bundle: {5911cf72-9c50-4662-a712-28a4c2f8b599}, from path: C:\ProgramData\Package Cache\{5911cf72-9c50-4662-a712-28a4c2f8b599}\
[1074:07F8][2019-11-26T13:28:54]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{5911cf72-9c50-4662-a712-28a4c2f8b599}, resume: None, restart initiated: No, disable resume: No
[1310:0624][2019-11-26T13:28:55]i399: Apply complete, result: 0x80070643, restart: None, ba requested restart:  No

I imagine the parameter INSTALLFOLDER="" is what causes it to fail, and as there's no way to run the extracted *.msi-installer outside the .exe-installer I have no way to specify the parameter.

**Comment by OttoBoll** [comment]
Before anyone asks:
PS C:\Windows\system32> $env:ProgramFiles
C:\Program Files

**Comment by OttoBoll** [comment]
Purely empirical observation: 
IIS.Administration install fails everywhere we have .Net Core 3.0.0 hosting bundle installed.
.Net Core 3.0.0 is not recognized by IIS.Administration pre-req installer as a compatible version of .Net Core.
Only place I've managed to successfully install IIS.Administration is on a "clean" server with no prior versions of .Net Core.

**Comment by bariscaglar** [comment]
@shhsu can you take a look?

**Comment by OttoBoll** [comment]
Can now confirm (albeit using a very small sample size) that the .Net Core 3.0.0 Hosting Bundle is the thing preventing a successful install.
.Net Core 3.0.0 Hosting Bundle can be successfully reinstalled afterwards to utilize functionality from both.

C:\Windows\system32>dotnet --info
  It was not possible to find any installed .NET Core SDKs
  Did you mean to run .NET Core SDK commands? Install a .NET Core SDK from:
      https://aka.ms/dotnet-download

Host (useful for support):
  Version: 3.0.0
  Commit:  7d57652f33

.NET Core SDKs installed:
  No SDKs were found.

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNe
tCore.All]
  Microsoft.AspNetCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNe
tCore.App]
  Microsoft.AspNetCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNe
tCore.App]
  Microsoft.NETCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.
App]
  Microsoft.NETCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.
App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download

**Comment by shhsu** [comment]
I have tried a successful installation and INSTALLFOLDER as empty on my side as well. This shouldn't be the direct reason why installer failed. Can you share the full Microsoft_IIS_Administration_2.3.0_*.log and Microsoft_IIS_Administration_2.3.0_*_iis_administration_msi.log files?

**Comment by OttoBoll** [comment]
[IISAdministration_Logs_SuccessfulInstall.zip](https://github.com/microsoft/IIS.Administration/files/3930970/IISAdministration_Logs_SuccessfulInstall.zip)
[IISAdministration_Logs_FailedInstall.zip](https://github.com/microsoft/IIS.Administration/files/3930971/IISAdministration_Logs_FailedInstall.zip)

**Comment by shhsu** [comment]
Somehow when the installer try to start the service, failure occurred and the installation got rolled back. Unfortunately there is not a general way to troubleshoot this situation. Can you observe the events generated in Event Viewer and see if there's anything noteworthy coming up during installations?


```
CAQuietExec64:  VERBOSE: Start-Service for Microsoft IIS Administration started.
CAQuietExec64:  Start-Service : Service 'Microsoft IIS Administration (Microsoft IIS Administration)' cannot be started due to the foll
CAQuietExec64:  owing error: Cannot start service Microsoft IIS Administration on computer '.'.
CAQuietExec64:  At C:\Program Files\IIS Administration\2.3.0\setup\install.ps1:60 char:17
CAQuietExec64:  +                 Start-Service -name $name
CAQuietExec64:  +                 ~~~~~~~~~~~~~~~~~~~~~~~~~
CAQuietExec64:      + CategoryInfo          : OpenError: (System.ServiceProcess.ServiceController:ServiceController) ,  
CAQuietExec64:     ServiceCommandException
CAQuietExec64:      + FullyQualifiedErrorId : CouldNotStartService,Microsoft.PowerShell.Commands.StartServiceCommand
CAQuietExec64:   
CAQuietExec64:  VERBOSE: Checking the current status. Status: Stopped
CAQuietExec64:  WARNING: Get-Service does not return Running. Status: Stopped
CAQuietExec64:  VERBOSE: Retrying to call Start-Service for Microsoft IIS Administration.
CAQuietExec64:  Start-Service : Service 'Microsoft IIS Administration (Microsoft IIS Administration)' cannot be started due to the foll
CAQuietExec64:  owing error: Cannot start service Microsoft IIS Administration on computer '.'.
CAQuietExec64:  At C:\Program Files\IIS Administration\2.3.0\setup\install.ps1:72 char:21
CAQuietExec64:  +                     Start-Service -name $name
CAQuietExec64:  +                     ~~~~~~~~~~~~~~~~~~~~~~~~~
CAQuietExec64:      + CategoryInfo          : OpenError: (System.ServiceProcess.ServiceController:ServiceController) ,  
CAQuietExec64:     ServiceCommandException
CAQuietExec64:      + FullyQualifiedErrorId : CouldNotStartService,Microsoft.PowerShell.Commands.StartServiceCommand
CAQuietExec64:   
CAQuietExec64:  VERBOSE: StartService failed. Retrying in 1 seconds. Error: Unexpected status: Stopped
CAQuietExec64:  VERBOSE: Checking the current status. Status: Stopped
CAQuietExec64:  WARNING: Get-Service does not return Running. Status: Stopped
CAQuietExec64:  VERBOSE: StartService failed. Retrying in 1 seconds. Error: Unexpected status: Stopped
CAQuietExec64:  VERBOSE: Checking the current status. Status: Stopped
CAQuietExec64:  WARNING: Get-Service does not return Running. Status: Stopped
CAQuietExec64:  WARNING: StartService failed the maximum number of 2 times. Error: Unexpected status: Stopped
CAQuietExec64:  Rolling back
CAQuietExec64:  Rolling back service creation
CAQuietExec64:  Rolling back HTTP.Sys port binding
CAQuietExec64:  VERBOSE: Deleting Ssl Cert Configuration
CAQuietExec64:  Rolling back setup config creation
CAQuietExec64:  Rolling back SSL certificate creation
CAQuietExec64:  Finished rolling back.
CAQuietExec64:  C:\Program Files\IIS Administration\2.3.0\setup\msi-setup.ps1 : Unexpected status: Stopped
CAQuietExec64:  At C:\Program Files\IIS Administration\2.3.0\setup\install.ps1:74 char:17
CAQuietExec64:  +                 throw ("Unexpected status: " + $status)
CAQuietExec64:  +                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
CAQuietExec64:  At line:1 char:1
CAQuietExec64:  + & 'C:\Program Files\IIS Administration\2.3.0\setup\msi-setup.ps1' Ins ...
CAQuietExec64:  + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
CAQuietExec64:      + CategoryInfo          : NotSpecified: (:) , RuntimeException
CAQuietExec64:      + FullyQualifiedErrorId : System.Management.Automation.RuntimeException,msi-setup.ps1
CAQuietExec64:  Error 0x80070001: Command line returned an error.
CAQuietExec64:  Error 0x80070001: QuietExec64 Failed
CAQuietExec64:  Error 0x80070001: Failed in ExecCommon64 method
```

**Comment by OttoBoll** [comment]
Here's the logs by category I can find that pertains to a failed setup/installation:

[Logs.zip](https://github.com/microsoft/IIS.Administration/files/3943629/Logs.zip)

These logs are gathered on a separate machine from the one I gathered these logs on https://github.com/microsoft/IIS.Administration/issues/291#issuecomment-562471850, as filtering information level events from that long ago proved a bit too laborious.
The install still fails with the same exit code and behaviour.

**Comment by OttoBoll** [comment]
I also noticed this "new" dotnet --info readout on the machine that I managed to install IIS.Administration on when comparing:

![image](https://user-images.githubusercontent.com/28365908/70507148-100cb280-1b2c-11ea-98bc-3d17d3780c87.png)

This is the same machine that 5 days ago had this readout:
https://github.com/microsoft/IIS.Administration/issues/291#issuecomment-562160084

**Comment by shhsu** [comment]
This is odd, can you go to `Services` and look at `Microsoft IIS Administration` and ensure its execution path is `C:\Program Files\dotnet\dotnet.exe "C:\Program Files\IIS Administration\2.3.0\Microsoft.IIS.Administration\Microsoft.IIS.Administration.dll" /serviceName="Microsoft IIS Administration"`
Also, can you share the console output for `"C:\Program Files\dotnet\dotnet.exe" --info`

**Comment by OttoBoll** [comment]
> This is odd, can you go to Services and look at Microsoft IIS Administration and ensure its execution path is C:\Program Files\dotnet\dotnet.exe "C:\Program Files\IIS Administration\2.3.0\Microsoft.IIS.Administration\Microsoft.IIS.Administration.dll" /serviceName="Microsoft IIS Administration"

C:\Program Files\dotnet\dotnet.exe "C:\Program Files\IIS Administration\2.3.0\Microsoft.IIS.Administration\Microsoft.IIS.Administration.dll" /serviceName="Microsoft IIS Administration"

> Also, can you share the console output for "C:\Program Files\dotnet\dotnet.exe" --info

![image](https://user-images.githubusercontent.com/28365908/72240952-d07f2d80-35e5-11ea-968b-533cf6bfd266.png)

System ENV VAR PATH:
C:\Program Files (x86)\Common Files\Oracle\Java\javapath;%SystemRoot%\system32;%SystemRoot%;%SystemRoot%\System32\Wbem;%SYSTEMROOT%\System32\WindowsPowerShell\v1.0\;E:\PHP\php-7.2.3-nts-Win32-VC15-x64\php-cgi.exe;C:\Program Files\Microsoft\Web Platform Installer\;C:\Program Files\k6\;C:\Program Files (x86)\dotnet\;C:\Program Files\dotnet\

**Comment by shhsu** [comment]
I have set up a vm. Installed only .net core 3.0 in it and was able to repro the issue. Somehow after windows update policy kicked in and rebooted the machine. I was able to install IIS Administration again and the issue has disappeared.
I am going to re-create the VM and see if I can repro the problem yet again. Meanwhile, just updating the thread for record

**Comment by OttoBoll** [comment]
Any updates?

**Comment by davidthompson** [comment]
Has anyone had any luck with this?

**Comment by shhsu** [comment]
@davidthompson 

I cannot repro it since

**Comment by MiamiFinsFan13** [comment]
I am having the same issue (see log) running on windows server 2012r2 with no .net core 3 installed. I've tried running regularly and as admin. 
`[2730:1B50][2021-09-13T09:09:44]i001: Burn v3.11.1.2318, Windows v6.3 (Build 9600: Service Pack 0), path: C:\Users\[redacted]\AppData\Local\Temp\13\{CDEB66C6-4127-4C19-A1CE-A631E95E1073}\.cr\IISAdministrationSetup.exe
[2730:1B50][2021-09-13T09:09:44]i000: Initializing version variable 'DOTNET_RUNTIME_VERSION_MINIMAL' to value '2.1.0'
[2730:1B50][2021-09-13T09:09:44]i000: Initializing string variable 'DefaultCors' to value 'true'
[2730:1B50][2021-09-13T09:09:44]i009: Command Line: '-burn.clean.room=C:\Users\j_kennedy_ta\Desktop\IISAdministrationSetup.exe -burn.filehandle.attached=304 -burn.filehandle.self=312'
[2730:1B50][2021-09-13T09:09:44]i000: Setting string variable 'WixBundleOriginalSource' to value 'C:\Users\j_kennedy_ta\Desktop\IISAdministrationSetup.exe'
[2730:1B50][2021-09-13T09:09:44]i000: Setting string variable 'WixBundleOriginalSourceFolder' to value 'C:\Users\j_kennedy_ta\Desktop\'
[2730:1B50][2021-09-13T09:09:44]i000: Setting string variable 'WixBundleLog' to value 'C:\Users\J_KENN~2\AppData\Local\Temp\Microsoft_IIS_Administration_2.3.0_20210913090944.log'
[2730:1B50][2021-09-13T09:09:44]i000: Setting string variable 'WixBundleName' to value 'Microsoft IIS Administration 2.3.0'
[2730:1B50][2021-09-13T09:09:44]i000: Setting string variable 'WixBundleManufacturer' to value 'Microsoft Corporation'
[2730:2060][2021-09-13T09:09:44]i000: Setting numeric variable 'WixStdBALanguageId' to value 1033
[2730:2060][2021-09-13T09:09:44]i000: Setting version variable 'WixBundleFileVersion' to value '2.3.0.0'
[2730:1B50][2021-09-13T09:09:44]i100: Detect begin, 3 packages
[2730:1B50][2021-09-13T09:09:44]i000: Setting numeric variable 'ASPNET_CORE_21_SHARED_FRAMEWORK_EXISTS' to value 1
[2730:1B50][2021-09-13T09:09:44]i000: Setting string variable 'DOTNET_RUNTIME_VERSION' to value '2.1.1'
[2730:1B50][2021-09-13T09:09:44]i052: Condition '(DOTNET_RUNTIME_VERSION) AND (DOTNET_RUNTIME_VERSION >= DOTNET_RUNTIME_VERSION_MINIMAL)' evaluates to true.
[2730:1B50][2021-09-13T09:09:44]i052: Condition '(ASPNET_CORE_21_SHARED_FRAMEWORK_EXISTS)' evaluates to true.
[2730:1B50][2021-09-13T09:09:44]i101: Detected package: dotnet_core_exe, state: Present, cached: None
[2730:1B50][2021-09-13T09:09:44]i101: Detected package: aspnetcore_exe, state: Present, cached: None
[2730:1B50][2021-09-13T09:09:44]i101: Detected package: iis_administration_msi, state: Absent, cached: Complete
[2730:1B50][2021-09-13T09:09:44]i199: Detect complete, result: 0x0
[2730:2060][2021-09-13T09:09:46]i000: Setting numeric variable 'EulaAcceptCheckbox' to value 1
[2730:1B50][2021-09-13T09:09:46]i200: Plan begin, 3 packages, action: Install
[2730:1B50][2021-09-13T09:09:46]w321: Skipping dependency registration on package with no dependency providers: dotnet_core_exe
[2730:1B50][2021-09-13T09:09:46]w321: Skipping dependency registration on package with no dependency providers: aspnetcore_exe
[2730:1B50][2021-09-13T09:09:46]i052: Condition 'VersionNT64 AND VersionNT >= v6.1' evaluates to true.
[2730:1B50][2021-09-13T09:09:46]i000: Setting string variable 'WixBundleRollbackLog_iis_administration_msi' to value 'C:\Users\[redacted]\AppData\Local\Temp\Microsoft_IIS_Administration_2.3.0_20210913090944_000_iis_administration_msi_rollback.log'
[2730:1B50][2021-09-13T09:09:46]i000: Setting string variable 'WixBundleLog_iis_administration_msi' to value 'C:\Users\[redacted]\AppData\Local\Temp\Microsoft_IIS_Administration_2.3.0_20210913090944_000_iis_administration_msi.log'
[2730:1B50][2021-09-13T09:09:46]i201: Planned package: dotnet_core_exe, state: Present, default requested: Present, ba requested: Present, execute: None, rollback: None, cache: No, uncache: No, dependency: None
[2730:1B50][2021-09-13T09:09:46]i201: Planned package: aspnetcore_exe, state: Present, default requested: Present, ba requested: Present, execute: None, rollback: None, cache: No, uncache: No, dependency: None
[2730:1B50][2021-09-13T09:09:46]i201: Planned package: iis_administration_msi, state: Absent, default requested: Present, ba requested: Present, execute: Install, rollback: Uninstall, cache: No, uncache: No, dependency: Register
[2730:1B50][2021-09-13T09:09:46]i299: Plan complete, result: 0x0
[2730:1B50][2021-09-13T09:09:46]i300: Apply begin
[2730:1B50][2021-09-13T09:09:46]i010: Launching elevated engine process.
[2730:1B50][2021-09-13T09:09:46]i011: Launched elevated engine process.
[2730:1B50][2021-09-13T09:09:46]i012: Connected to elevated engine.
[0570:1C40][2021-09-13T09:09:46]i358: Pausing automatic updates.
[0570:1C40][2021-09-13T09:09:46]i359: Paused automatic updates.
[0570:1C40][2021-09-13T09:09:46]i360: Creating a system restore point.
[0570:1C40][2021-09-13T09:09:46]i362: System restore disabled, system restore point not created.
[0570:1C40][2021-09-13T09:09:46]i370: Session begin, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{5911cf72-9c50-4662-a712-28a4c2f8b599}, options: 0x7, disable resume: No
[0570:1C40][2021-09-13T09:09:46]i000: Caching bundle from: 'C:\Users\[redacted]\AppData\Local\Temp\13\{15AA2F63-8152-40FF-BE1F-62B22DF9DB78}\.be\IISAdministrationSetup.exe' to: 'C:\ProgramData\Package Cache\{5911cf72-9c50-4662-a712-28a4c2f8b599}\IISAdministrationSetup.exe'
[0570:1C40][2021-09-13T09:09:46]i320: Registering bundle dependency provider: {5911cf72-9c50-4662-a712-28a4c2f8b599}, version: 2.3.0.0
[0570:1C40][2021-09-13T09:09:46]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{5911cf72-9c50-4662-a712-28a4c2f8b599}, resume: Active, restart initiated: No, disable resume: No
[0570:1FC0][2021-09-13T09:09:46]i304: Verified existing payload: iis_administration_msi at path: C:\ProgramData\Package Cache\{51B98E36-BBAC-4549-8460-DBEFFE4FDD06}v2.3.0\packages\iis_administration.msi.
[0570:1C40][2021-09-13T09:09:46]i323: Registering package dependency provider: {51B98E36-BBAC-4549-8460-DBEFFE4FDD06}, version: 2.3.0, package: iis_administration_msi
[0570:1C40][2021-09-13T09:09:46]i301: Applying execute package: iis_administration_msi, action: Install, path: C:\ProgramData\Package Cache\{51B98E36-BBAC-4549-8460-DBEFFE4FDD06}v2.3.0\packages\iis_administration.msi, arguments: ' ARPSYSTEMCOMPONENT="1" MSIFASTINSTALL="7" INSTALLFOLDER="" VSEXTUI="1" DEFAULT_CORS="true"'
[0570:1C40][2021-09-13T09:09:52]e000: Error 0x80070643: Failed to install MSI package.
[0570:1C40][2021-09-13T09:09:52]e000: Error 0x80070643: Failed to execute MSI package.
[2730:1B50][2021-09-13T09:09:52]e000: Error 0x80070643: Failed to configure per-machine MSI package.
[2730:1B50][2021-09-13T09:09:52]i319: Applied execute package: iis_administration_msi, result: 0x80070643, restart: None
[2730:1B50][2021-09-13T09:09:52]e000: Error 0x80070643: Failed to execute MSI package.
[0570:1C40][2021-09-13T09:09:52]i372: Session end, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{5911cf72-9c50-4662-a712-28a4c2f8b599}, resume: None, restart: None, disable resume: No
[0570:1C40][2021-09-13T09:09:52]i330: Removed bundle dependency provider: {5911cf72-9c50-4662-a712-28a4c2f8b599}
[0570:1C40][2021-09-13T09:09:52]i352: Removing cached bundle: {5911cf72-9c50-4662-a712-28a4c2f8b599}, from path: C:\ProgramData\Package Cache\{5911cf72-9c50-4662-a712-28a4c2f8b599}\
[0570:1C40][2021-09-13T09:09:52]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{5911cf72-9c50-4662-a712-28a4c2f8b599}, resume: None, restart initiated: No, disable resume: No
[2730:1B50][2021-09-13T09:09:52]i399: Apply complete, result: 0x80070643, restart: None, ba requested restart:  No
`

**Comment by trontronicent** [comment]
i was able top get out of this situation.

Originally i tried to install 2.3.0 with having just net48 and net6 sdks / runtimes installed on Server 2022.
So i decided to try by adding the latest NETCore 2.* and NETCore 3.*, just for the purpose of getting the install to work. 

After that I could install the 2.3.0 directly through WAC, without any issues.

**Comment by yaqiyang** [comment]
The installer in 2.3.0 tryies to install older versions of .net. It fails when a newer version exists, and the installer will then fail. This has been fixed in v6.0
